course_info/tests/jasmine/controllers/DetailsControllerSpec.js

```diff
@@ -0,0 +1,237 @@
+                    school_id: 'abc',
+                    registrar_code_display: '2222',
+                    course_id: '789',
+                    course_groups: [
+                        {
+                            name: 'group one',
+                            id: 1
+                        },
+                        {
+                            name: 'group two',
+                            id: 2
+                        }
+                    ],
+                    departments : [
+                        {
+                            name: 'dept1',
+                            id: 1
+                        }
+                    ]
+                },
+                term: {
+                    display_name: 'Summer 2015',
+                    academic_year: '2015',
+                },
+                primary_xlist_instances: [],
+            };
+
+            members = {
+                count: 100
+            };
+        });
+
+        it('format the course instance data for the UI ', function () {
+            courseInstances.instances[ci.course_instance_id] = ci;
+            dc = $controller('DetailsController', {$scope: scope});
+            $httpBackend.expectGET(courseInstanceURL).respond(200, JSON.stringify(ci));
+            $httpBackend.expectGET(peopleURL).respond(200, JSON.stringify({}));
+            $httpBackend.flush(2);
+            expect(dc.courseInstance['title']).toEqual(ci.title);
+            expect(dc.courseInstance['short_title']).toEqual(ci.short_title);
+            expect(dc.courseInstance['sub_title']).toEqual(ci.sub_title);
+            expect(dc.courseInstance['school']).toEqual
+            (ci.course.school_id.toUpperCase());
+            expect(dc.courseInstance['term']).toEqual(ci.term.display_name);
+            expect(dc.courseInstance['course_groups']).toEqual(ci.course.course_groups);
+            expect(dc.courseInstance['departments']).toEqual(ci.course.departments);
+            expect(dc.courseInstance['year']).toEqual(ci.term.academic_year);
+            expect(dc.courseInstance['cid']).toEqual(ci.course_instance_id);
+            expect(dc.courseInstance['registrar_code_display']).toEqual(
+                ci.course.registrar_code_display + ' (' + ci.course.course_id + ')');
+            expect(dc.courseInstance['sites']).toEqual(ci.sites);
+            expect(dc.courseInstance['xlist_status']).toEqual('N/A');
+            expect(dc.courseInstance['description']).toEqual('<p>hello</p>');
+            // members.count is not defined it should equal 0
+            expect(dc.courseInstance['members']).toEqual(0);
+
+        });
+
+        it('should deal with an empty ci', function () {
+            dc = $controller('DetailsController', {$scope: scope});
+            $httpBackend.expectGET(courseInstanceURL).respond(200, JSON.stringify({}));
+            $httpBackend.expectGET(peopleURL).respond(200, JSON.stringify({}));
+            $httpBackend.flush(2);
```

Comment:
Do we log an error if the response is empty?  If so, we should be checking for it here.